docs/proposals/20220825-azuremanaged-cluster-exp-graduation.md

```diff
@@ -0,0 +1,126 @@
+
+## Issue Tracking
+
+Issues that emerge as a result of this effort will be tracked here:
+
+- https://github.com/orgs/kubernetes-sigs/projects/26
+
+## Motivation
+
+### Goals
+- Agree upon a durable, post-experimental specification of the set of CRDs that implement Managed Kubernetes on Azure, e.g., `AzureManagedCluster`, `AzureManagedControlPlane`, `AzureManagedMachinePool`
+- Prioritize timeliness of a post-experimental definition so that users can confidently plan for any breaking changes that result from a graduated spec as soon as possible
+- Prioritize "architectural affinity" with other Cluster API providers implementing Managed Kubernetes
+
+### Non-Goals / Future Work
+- Add to, or remove features from, "`AzureManagedCluster`"
+- Standardize any opinions about how best to use AKS
+- Promote CAPZ-specific opinions about Managed Kubernetes across the Cluster API provider ecosystem
+- Define operational support
+
+## Post-graduation Prerequisites
+
+Concrete prereq workstreams are [tracked here](https://github.com/orgs/kubernetes-sigs/projects/26/views/1)
+
+### 1. Land Managed Cluster in Cluster API Proposal (Status: DONE)
+
+See:
+
+- https://github.com/kubernetes-sigs/cluster-api/pull/6988
+
+The above proposal defines a set of recommendations for Cluster API providers implementing Managed Kubernetes solutions. Because this proposal is an opt-in collection of architectural recommendations, it is not required that "`AzureManagedCluster`" strictly agrees with everything therein. However, by contributing to the successful landing of that proposal in Cluster API, we can best ensure that any CAPZ-specific opinions or learnings are reflected, for the benefit of the larger ecosystem, and for the maximum happiness of AKS customers in particular.
+
+Ref:
+
+- https://github.com/kubernetes-sigs/cluster-api/pull/6988
+
+### 2. Consider Cluster API Proposal Recommendations (Status: Finalizing Consensus)
+
+Once the proposal is accepted and merged into the Cluster API project as an endorsed set of provider recommendations, CAPZ can audit the existing "`AzureManagedCluster`" experimental implementation for areas of disagreement with said recommendations. For each discovery of disagreement, there should be a defensible reason to matriculate a divergent CAPZ Managed Kubernetes implementation into a post-experimental definition of "`AzureManagedCluster`", and ideally some form of sign-off from other provider maintainers. Where community consensus cannot be reached, CAPZ should strongly consider evolving the experimental "`AzureManagedCluster`" implementation to meet the Cluster API Managed Kubernetes recommendation as a pre-requisite to graduating from experimental.
+
+Ref:
+
+- https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/2739
+- https://github.com/kubernetes-sigs/cluster-api/issues/7494
+
+
+### 3. Pathway Towards Full AKS Feature Support (Status: DONE)
```

Comment:
@nojnhuh @mtougeron @sonasingh46 @zioproto @LochanRn @luthermonson 

All of you (and apologies to others I haven't mentioned!) have made significant feature contributions to the AzureManagedCluster API recently. As a result of that effort, does anyone have any concerns that the existing API surface area may be non-durable w/ respect to graduation from experimental?

The plan is to "move the existing API forward", based on the conclusion that we have done sufficient validation to confidently predict that additional feature work can continue to successfully land on this existing API. In other words, we have no reason to believe that we'll have to break the API in order to accommodate future AKS features.

Can folks weigh in on their confidence level of the above? Especially if you feel otherwise!

Thank you 🙏

Reply:
One of the more potentially disruptive changes I can think of that we haven't addressed yet is exposing the AKS preview APIs (#2625). Even that seems like something we can enable in a backwards-compatible manner though. Overall, nothing else comes to mind that might threaten the integrity of the API based on where I've been focused recently.